Setup devenv on Windows 7 x64 behind proxy server

## Description

I am trying to setup devenv as it is described in [here](https://github.com/hyperledger/fabric/blob/master/docs/dev-setup/devenv.md). Everything work as expected except in case when the host I am running vagrant on is behind proxy server. The provisioning process fails with error that Docker can't pull busybox image from remote server.
## Describe How to Reproduce
1. No VMs, everything is clean
2. Configure Windows 7 x64 (env. variables, required software ...) as described on the devenv page.
3. Run `vagrant up` -> Get first error that Docker can't pull busybox image. I believe this is normal, because Docker is being installed during provision step.
4. Run `vagrant provision` -> Get second error that Docker again can't pull busybox image. Before I get this error, I saw that proxy server was configured for Docker. Logs below.

```
D:\go_src\src\github.com\hyperledger\fabric\devenv>vagrant provision
==> default: Configuring proxy for Apt...
==> default: Configuring proxy for Docker...
==> default: Configuring proxy environment variables...
==> default: Configuring proxy for npm...
==> default: Running provisioner: shell...
    default: Running: inline script
==> default: stdin: is not a tty
==> default: ++ export DOCKER_STORAGE_BACKEND=
==> default: ++ DOCKER_STORAGE_BACKEND=
==> default: ++ cd /hyperledger/devenv
==> default: ++ ./setup.sh
==> default: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: apt-transport-https is already the newest version.
==> default: ca-certificates is already the newest version.
==> default: 0 upgraded, 0 newly installed, 0 to remove and 61 not upgraded.
==> default: Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --homedir /tmp/tmp.3tpe7XUzq7 --no-auto-check-trustdb --trust-mo
del always --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E8
9F3A912897C070ADBF76221572C52609D
==> default: gpg:
==> default: requesting key 2C52609D from hkp server p80.pool.sks-keyservers.net
==> default: gpg:
==> default: key 2C52609D: "Docker Release Tool (releasedocker) <docker@docker.com>" not changed
==> default: gpg:
==> default: Total number processed: 1
==> default: gpg:
==> default:              unchanged: 1
==> default: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: apparmor is already the newest version.
==> default: linux-image-extra-3.13.0-85-generic is already the newest version.
==> default: docker-engine is already the newest version.
==> default: 0 upgraded, 0 newly installed, 0 to remove and 61 not upgraded.
==> default: docker stop/waiting
==> default: docker start/running, process 10221
==> default: Unable to find image 'busybox:latest' locally
==> default: Pulling repository docker.io/library/busybox
==> default: docker: Network timed out while trying to connect to https://index.docker.io/v1/repositories/library/busybox/images. You may want to che
ck your internet connection or if you are behind a proxy..
==> default: See 'docker run --help'.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Hopefully, I fixed issue with /etc/default/docker. If replace `echo "..." > /etc/default/docker` with the following:

```
DOCKER_OPTS="-s=${DOCKER_STORAGE_BACKEND_STRING} -r=true --api-cors-header='*' -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock ${DOCKER_OPTS}"
sed -i.bak '/^DOCKER_OPTS=/{h;s|=.*|=\"'"${DOCKER_OPTS}"'\"|};${x;/^$/{s||DOCKER_OPTS=\"'"${DOCKER_OPTS}"'\"|;H};x}' /etc/default/docker
```

proxy variables set by Vagrant no longer lost.

However, later when Docker builds image, the following error appears. Looks like that it is needed to setup proxy inside the containers as well.

```
==> default: Step 11 : RUN /hyperledger/scripts/provision/common.sh
==> default:  ---> Running in d4b0eec20014
==> default: W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/InRelease
==> default:
==> default: W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/InRelease
==> default:
==> default: W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-security/InRelease
==> default:
==> default: W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty/Release.gpg  Cannot initiate the connection to archive.ubuntu.com:80 (2
001:67c:1560:8001::11). - connect (101: Network is unreachable) [IP: 2001:67c:1560:8001::11 80]
==> default:
==> default: W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/Release.gpg  Cannot initiate the connection to archive.ubuntu.c
om:80 (2001:67c:1560:8001::11). - connect (101: Network is unreachable) [IP: 2001:67c:1560:8001::11 80]
==> default:
==> default: W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-security/Release.gpg  Cannot initiate the connection to archive.ubuntu.
com:80 (2001:67c:1560:8001::11). - connect (101: Network is unreachable) [IP: 2001:67c:1560:8001::11 80]
==> default:
==> default: W: Some index files failed to download. They have been ignored, or old ones used instead.
==> default:
==> default: Reading package lists...
==> default: Building dependency tree...
==> default:
==> default: Reading state information...
==> default: E:
==> default: Unable to locate package default-jre
==> default:
==> default:
==> default: The command '/bin/sh -c /hyperledger/scripts/provision/common.sh' returned a non-zero code: 100
==> default: make: *** [.baseimage-dummy] Error 1
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

```

--- a/devenv/setup.sh
+++ b/devenv/setup.sh
@@ -70,7 +70,8 @@ esac
 apt-get install -y linux-image-extra-$(uname -r) apparmor docker-engine
 
 # Configure docker
-echo "DOCKER_OPTS=\"-s=${DOCKER_STORAGE_BACKEND_STRING} -r=true --api-cors-header='*' -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock ${DOCKER_OPTS}\"" > /etc/default/docker
+DOCKER_OPTS="-s=${DOCKER_STORAGE_BACKEND_STRING} -r=true --api-cors-header='*' -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock ${DOCKER_OPTS}"
+sed -i.bak '/^DOCKER_OPTS=/{h;s|=.*|=\"'"${DOCKER_OPTS}"'\"|};${x;/^$/{s||DOCKER_OPTS=\"'"${DOCKER_OPTS}"'\"|;H};x}' /etc/default/docker
 
 service docker restart
 usermod -a -G docker vagrant # Add vagrant user to the docker group

--- a/scripts/provision/docker.sh
+++ b/scripts/provision/docker.sh
@@ -99,6 +99,12 @@ COPY scripts $REMOTESCRIPTS
 RUN $REMOTESCRIPTS/common.sh
 EOF
 
-docker build -t $NAME:latest $TMP
+[ ! -z "$http_proxy" ] && DOCKER_ARGS_PROXY="$DOCKER_ARGS_PROXY --build-arg http_proxy=$http_proxy"
+[ ! -z "$https_proxy" ] && DOCKER_ARGS_PROXY="$DOCKER_ARGS_PROXY --build-arg https_proxy=$https_proxy"
+[ ! -z "$HTTP_PROXY" ] && DOCKER_ARGS_PROXY="$DOCKER_ARGS_PROXY --build-arg HTTP_PROXY=$HTTP_PROXY"
+[ ! -z "$HTTPS_PROXY" ] && DOCKER_ARGS_PROXY="$DOCKER_ARGS_PROXY --build-arg HTTPS_PROXY=$HTTPS_PROXY"
+[ ! -z "$no_proxy" ] && DOCKER_ARGS_PROXY="$DOCKER_ARGS_PROXY --build-arg no_proxy=$no_proxy"
+[ ! -z "$NO_PROXY" ] && DOCKER_ARGS_PROXY="$DOCKER_ARGS_PROXY --build-arg NO_PROXY=$NO_PROXY"
+docker build $DOCKER_ARGS_PROXY -t $NAME:latest $TMP
 
 rm -rf $TMP

